### PR TITLE
fix(ec2): instance may not have a name

### PR DESCRIPTION
**Pull request by Hweinstock**
## Problem
Currently the toolkit is unable to load EC2 instances that are not assigned a name. It causes it to fail to load all of the instances associated with a specific region.
## Solution
- displayed as `(no name)` in QuickPick.
- displayed with `instanceId` in explorer.
- behavior above captured in test suite. 

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/ec2/explorer/ec2InstanceNode.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class Ec2InstanceNode extends AWSTreeNodeBase implements AWSResourceNode
     }
 
     public get name(): string {
-        return getNameOfInstance(this.instance) ?? '(no name)'
+        return getNameOfInstance(this.instance) ?? this.InstanceId
```
Comment (justinmk3) [inline src/ec2/explorer/ec2InstanceNode.ts:40]: Let's try this:
```suggestion
        return getNameOfInstance(this.instance) ?? `${this.InstanceId} (no name)`
```

File: src/test/shared/clients/defaultEc2Client.test.ts [hunk 2]
```diff
@@ -93,6 +93,97 @@ describe('extractInstancesFromReservations', function () {
                 actualResult
             )
         })
+
+    it('is able to process results without complete Tag field.', async function () {
```
Comment (justinmk3) [inline src/test/shared/clients/defaultEc2Client.test.ts:97]: "should", "is able", "correctly" are usually redundant in test descriptions.
 
```suggestion
    it('processes results without complete Tag field.', async function () {
```

File: src/test/shared/clients/defaultEc2Client.test.ts [hunk 3]
```diff
@@ -93,6 +93,97 @@ describe('extractInstancesFromReservations', function () {
...
+                { InstanceId: 'id2' },
+                { InstanceId: 'id3', name: 'name3', Tags: [{ Key: 'Name', Value: 'name3' }] },
+                { InstanceId: 'id4', Tags: [] },
+            ],
+            actualResult
+        )
+    })
+})
+
+describe('getSingleInstanceFilter', function () {
```
Comment (justinmk3) [inline src/test/shared/clients/defaultEc2Client.test.ts:138]:
```suggestion
describe('getInstancesFilter', function () {
```

File: src/test/shared/clients/defaultEc2Client.test.ts [hunk 4]
```diff
@@ -93,6 +93,97 @@ describe('extractInstancesFromReservations', function () {
...
+    const client = new Ec2Client('')
+
+    it('returns proper filter when given instanceId', function () {
+        const testInstanceId1 = 'test'
```
Comment (justinmk3) [inline src/test/shared/clients/defaultEc2Client.test.ts:142]: is this test duplicate of line 189?

File: src/test/shared/clients/defaultEc2Client.test.ts [hunk 5]
```diff
@@ -93,6 +93,97 @@ describe('extractInstancesFromReservations', function () {
...
+                Values: [testInstanceId1, testInstanceId2],
+            },
+        ]
+
+        assert.deepStrictEqual(expectedFilters2, actualFilters2)
+    })
+})
+
+describe('instanceHasName', function () {
+    it('correctly returns whether or not there is name attached to instance.', function () {
```
Comment (justinmk3) [inline src/test/shared/clients/defaultEc2Client.test.ts:167]:
```suggestion
    it('returns whether or not there is name attached to instance', function () {
```

**Review by justinmk3 — APPROVED** [review]

File: src/test/shared/clients/defaultEc2Client.test.ts (hunk 4 above)
Comment (Hweinstock) [reply src/test/shared/clients/defaultEc2Client.test.ts:142]: Yep, good catch. Sometimes when I merge master back in I could weird stuff like this, but I missed this one.

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**